### PR TITLE
Add options to TextDumpOutputer

### PR DESCRIPTION
**Pull request by Dr15Jones**
Added perEvent and summary options.

The newly added unit tests pass.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
@sabasehrish @knoepfel please review

File: TextDumpOutputer.h [hunk 1]
```diff
@@ -7,19 +7,24 @@
 namespace cce::tf {
 class TextDumpOutputer : public OutputerBase {
  public:
- TextDumpOutputer() {}
+ TextDumpOutputer(bool perEventDump, bool summaryDump): perEventDump_(perEventDump), summaryDump_(summaryDump), eventCount_{0} {}
```
Comment (knoepfel) [inline TextDumpOutputer.h:10]: `eventCount_` should be initialized before `perEventDump_`.

File: TextDumpOutputer.h [hunk 2]
```diff
@@ -7,19 +7,24 @@
 namespace cce::tf {
 class TextDumpOutputer : public OutputerBase {
```
Comment (knoepfel) [inline TextDumpOutputer.h:8]: There are a lot of overrides marked `final`.  Would it be easier to just mark the entire class as `final`?

```cpp
class TextDumpOutputer final : public OutputBase { ... };
```

File: TextDumpOutputer.h (hunk 1 above)
Comment (Dr15Jones) [reply TextDumpOutputer.h]: curious that this wasn't caught by the compiler. Guess we need different warning settings?

File: TextDumpOutputer.h (hunk 2 above)
Comment (Dr15Jones) [reply TextDumpOutputer.h]: done

**Pull request merged**